File: 11726_2xN_타일링/main.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <math.h>

#define fast() cin.tie(NULL), ios::sync_with_stdio(false)
using namespace std;
long long d[10001];

long long dp(int n) {
    d[1] = 1;
    d[2] = 2;
    for(int i = 3; i <=n; i++){
        d[i] = (d[i - 1] + d[i - 2]) % 10007;
    }
    return d[n];
}

int main() {
    fast();
    int n;
    cin >> n;
    cout << dp(n);
}

File: 11726_2xN_타일링/main.py
import math

d = [0 for _ in range(1001)]

def dp(n):
    d[1] = int(1)
    d[2] = int(2)
    for i in range(3, n + 1):
        d[i] = (d[i - 2] + d[i - 1]) % 10007
    return d[n]

print(dp(int(input())))
